### PR TITLE
DS-687 | @ericbakenhus | Fix lat and long

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Storyblok represents number fields as strings in JSON (WHY?!?) so we need to convert lat and long values for Algolia

# Review By (Date)
- Soon/Retro

# Criticality
- Medium

# Review Tasks

## Setup tasks and/or behavior to test

1. Look at code

# Associated Issues and/or People
- DS-687

### CODE REVIEW

File: src/utilities/synchronizedEvents.js [hunk 1]
```diff
@@ -68,13 +68,9 @@ export const storyToAlgoliaEvent = (story) => {
...
-  const geo =
-    mergedEventData.latitude && mergedEventData.longitude
-      ? {
-          lat: mergedEventData.latitude,
-          lng: mergedEventData.longitude,
-        }
-      : null;
+  const lat = parseFloat(mergedEventData.latitude) || null;
+  const lng = parseFloat(mergedEventData.longitude) || null;
+  const geo = lat && lng ? { lat, lng } : null;
```
Comment (ericbakenhus) [inline src/utilities/synchronizedEvents.js:73]: 😐

File: src/utilities/synchronizedEvents.js [hunk 2]
```diff
@@ -68,13 +68,11 @@ export const storyToAlgoliaEvent = (story) => {
...
-      ? {
-          lat: mergedEventData.latitude,
-          lng: mergedEventData.longitude,
-        }
-      : null;
+  const lat = parseFloat(mergedEventData.latitude);
+  const lng = parseFloat(mergedEventData.longitude);
+  const hasValidLat = !!lat || lat === 0;
+  const hasValidLng = !!lng || lng === 0;
+  const geo = hasValidLat && hasValidLng ? { lat, lng } : null;
```
Comment (ericbakenhus) [inline src/utilities/synchronizedEvents.js:75]: 😐

Comment (ericbakenhus) [reply src/utilities/synchronizedEvents.js:75]: Min and max values are enforced within Storyblok, so we don't care about that here.

File: package.json [hunk 3]
```diff
@@ -72,7 +72,7 @@
   ],
   "license": "MIT",
   "scripts": {
-    "build": "gatsby build",
+    "build": "gatsby clean && gatsby build",
```
Comment (ericbakenhus) [inline package.json]: Dangit Gatsby...

File: src/utilities/synchronizedEvents.js [hunk 4]
```diff
@@ -68,13 +68,11 @@ export const storyToAlgoliaEvent = (story) => {
...
     : null;
-  const geo =
-    mergedEventData.latitude && mergedEventData.longitude
-      ? {
-          lat: mergedEventData.latitude,
-          lng: mergedEventData.longitude,
-        }
-      : null;
+  const lat = parseFloat(mergedEventData.latitude);
+  const lng = parseFloat(mergedEventData.longitude);
```
Comment (sherakama) [inline src/utilities/synchronizedEvents.js:72]: Everything is a string, except when it isn't.

**Pull request merged**